### PR TITLE
Copy the start of the indexer code into a separate package.

### PR DESCRIPTION
**Pull request by silasbw**
### Description

This is copy-paste from @gnardini's indexer branch at
https://github.com/celo-org/wallet/tree/indexer. The indexing code in
that branch is built in the notification-service package. We plan to
eventually refactor the notification fuctionality out of that package,
but until we do it's difficult to collaborate on the indexer code.

To facilitate immediate collaboration, we're copy-pasting the inter-mixed
indexer+notification-service code into a new package. A separate follow-on
PR will remove the notification functionality from the indexer package.

### CODE REVIEW

File: packages/indexer/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Notification Service
+
+This service is responsible for dispatching notifications to mobile clients.
```
Comment (jeanregisser) [inline packages/indexer/README.md:3]: We should mention `indexer` instead.

File: packages/indexer/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+echo 'Copying over config file'
+cp "config/config.${NETWORK}.env" .env
+
+echo 'Deploying to gcloud'
+gcloud --project ${PROJECT} app deploy -q "app.${NETWORK}.yaml"
+
+echo 'Hitting service url to trigger update'
+# This seems to be necessary to ensure get App Engine starts the service
+curl "https://notification-dot-${PROJECT}.appspot.com" > /dev/null 2>&1
```
Comment (jeanregisser) [inline packages/indexer/deploy.sh:32]: Will probably be:
```suggestion
curl "https://indexer-dot-${PROJECT}.appspot.com" > /dev/null 2>&1
```

File: packages/indexer/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    n) NETWORK="$OPTARG" ;;
+    *) error "Unexpected option ${flag}" ;;
+  esac
+done
+
+[ -z "$NETWORK" ] && echo "Need to set the NETWORK via the -n flag" && exit 1;
+
+PROJECT="celo-mobile-${NETWORK}"
+
+echo "Starting notification service deployment."
```
Comment (jeanregisser) [inline packages/indexer/deploy.sh:22]:
```suggestion
echo "Starting indexer service deployment."
```

File: packages/indexer/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# Deploys the notification serivice to App Engine
```
Comment (jeanregisser) [inline packages/indexer/deploy.sh:4]:
```suggestion
# Deploys the indexer service to App Engine
```

**Review by jeanregisser — APPROVED** [review]
LGTM 👍

**Comment by silasbw** [comment]
Thanks @jeanregisser .  @gnardini also encouraged me to clear out some of the notification-service code as part of this PR to make the indexer code easier to review.  I'll take a pass at that and solicit another round of feedback.

File: packages/indexer/src/util/utils.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import { CeloContract, ContractKit, newKitFromWeb3 } from '@celo/contractkit'
+import Web3 from 'web3'
+import { WEB3_PROVIDER_URL } from '../config'
+
+interface ObjectWithStringsAndUndefined {
```
Comment (gnardini) [inline packages/indexer/src/util/utils.ts:5]: nit: Most of the things in this file I thing aren't being used

**Review by gnardini — APPROVED** [review]
🚀  Happy to see this going live :D

File: packages/indexer/src/util/utils.ts (hunk 5 above)
Comment (silasbw) [reply packages/indexer/src/util/utils.ts]: removed!

**Pull request merged**